zee_slider.js

```diff
@@ -121,9 +124,9 @@
       this.bulletsContainer.className = "zeeSlider-bullets-container";
       for (i=0; i < this.imagesCount; i++) {
         if ( i === 0 ) {
-          this.bulletsContainer.innerHTML +=  "<span class='active bullet_"+ i +"'>o</span>";
+          this.bulletsContainer.innerHTML +=  "<span class='active' data-bullet-index='"+ i +"'>o</span>";
```

Comment:
this is a bit less effecient, as innerHTML is very cpu-expensive.
I would suggest to one thing... 
create a `documentFragment`, give it a class `zeeSlider-bullets-container` then use `document.creatElement` and creat bullets and append to the container.

then replace the docmentFragment with the current container in the dom. 
with that you gained to major benefits
1. didn't use innerHTML :) 
2. only inserted in the dom once, which basically better than appending into it several times in loop.

Reply:
@Mohamed-amin This is what the buildOut method does. It creates all elements of the slider in a docFrag and after that happens, the docFrag is appended to the DOM.
Do you mean even inner elements in the docFrag such as the bullets should be added in another docFrag as well, then appended to the parent docFrag?

Reply:
oh, sorry my bad I didn't see the above code :)